lib/java_buildpack/framework/checkmarx_iast_agent.rb

```diff
@@ -0,0 +1,94 @@
+# frozen_string_literal: true
+
+# Cloud Foundry Java Buildpack
+# Copyright 2013-2020 the original author or authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'java_buildpack/framework'
+
+module JavaBuildpack
+  module Framework
+
+    # Encapsulates the functionality for running with Checkmarx IAST Agent
+    class CheckmarxIastAgent < JavaBuildpack::Component::BaseComponent
+      include JavaBuildpack::Util
+
+      # Creates an instance.  In addition to the functionality inherited from +BaseComponent+, +@version+ and +@uri+
+      # instance variables are exposed.
+      #
+      # @param [Hash] context a collection of utilities used by components
+      def initialize(context)
+        super(context)
+
+        # Save the IAST server URL in server, if found
+        service = @application.services.find_service(FILTER, 'server')
+        @server = service['credentials']['server'].chomp '/' if service
+      end
+
+      # (see JavaBuildpack::Component::BaseComponent#detect)
+      def detect
+        @server
+      end
+
+      # (see JavaBuildpack::Component::BaseComponent#compile)
+      def compile
+        # Download and extract the agent from the IAST server
+        FileUtils.mkdir_p @droplet.sandbox
+        # curl --insecure: most IAST servers will use self-signed SSL
+        shell 'curl --fail --insecure --silent --show-error ' \
```

Comment:
In general we want to stay away from calling the shell and hence having a dependency on the operating system, unless we absolutely need it.  In this case we'll use the Ruby libraries (including utilities within the buildpack) to download and expand this file.